CS410_Assignment2/src/SensorApplication/sensor/SensorAdapter.java

```diff
@@ -0,0 +1,34 @@
+/* SensorAdapter is an abstract class that contains all the fields and methods shared 
+ * by all of its derived classes: TemperatureSensorAdapter, PressureSensorAdapter, RadiationSensorAdapter.
+ */
+
+package sensor;
+
+import java.awt.Color;
+
+public abstract class SensorAdapter implements Sensor{
+
+	@Override
+	public String getType() {
+		return type;
+	}
+	
+	@Override
+	public Color getStatusColor() {
+		if(status == "OK")
+			return Color.GREEN;
+		else if(status == "CRITICAL")
+			return Color.YELLOW;
+		else
+			return Color.RED;
+	}

```

Comment:
1. I do not think it is right to handle UI specific logic in the Adapter. The Adapter should just return the status and the UI layer should handle what color it wants to map to the status value.
2. This is also assuming that all adaptee's will return the same set of status values. It will be a problem if in future, one of the adaptees choose to return a different set of values while others continue to return existing values.